Objects/genericaliasobject.c

```diff
@@ -225,6 +226,33 @@ tuple_add(PyObject *self, Py_ssize_t len, PyObject *item)
     return 0;
 }
 
+// Makes a shallow copy of a tuple
+static inline PyObject *
+tuple_copy(PyObject *obj) {
+    return _PyTuple_FromArray(((PyTupleObject *)obj)->ob_item, Py_SIZE(obj));
+}
+
+// This converts all nested lists in args to a tuple (args should be a tuple).
+// Usually needed to help caching in other libraries.
+// Must be called on a copy of a tuple object, where the only reference is
+// owned by the caller.
+static inline void
+tupleify_lists(PyObject *args) {
+    Py_ssize_t len = PyTuple_GET_SIZE(args);
+    for (Py_ssize_t i = 0; i < len; ++i) {
+        PyObject *arg = PyTuple_GET_ITEM(args, i);
+        int is_list = PyList_CheckExact(arg);
+        int is_tuple = PyTuple_CheckExact(arg);
+        if (is_list || is_tuple) {
+            // In case there are lists nested inside tuples.
+            PyObject *new_arg = is_list ? PyList_AsTuple(arg) : tuple_copy(arg);
```

Comment:
You need to check if `new_arg` is `NULL`. Both `PyList_AsTuple()` and `tuple_copy()` can fail and return `NULL`.